### PR TITLE
Improve naming

### PR DESCRIPTION
**Pull request by ageorgou**
A more structured attempt to make naming of resources, URLs etc smoother and consistent (this should replace #49). Mostly this moves some things around and separates the "base name" (given in the form) from the "DNS name" (the domain actually used, including random tokens to avoid clashes). Both are stored in the DB so they can be used later on (e.g. in assigning the UCL domain), and a new names module will contain relevant functions for generating names.

Note: This probably needs the migration code for adding a column to the DB schema; the default generated upgrade will fail if there are any rows already in the DB, as they will violate the non-null constraint. Editing the migration is easy, but I'm not sure what we're assuming for version-controlling the DB contents/migrations.

Other things to possibly implement (but maybe not in this PR?):
- validation of the given base name (although "unacceptable" characters are removed at the moment)
- checks for uniqueness of names

### CODE REVIEW

**Comment by ageorgou** [comment]
Tests seem to fail because they use syntax introduced in Python 3.5 and the tests run in 3.4. Sorry! I can change it - I assume we do want to test 3.4 too?

**Comment by jonc125** [comment]
The generated (and possibly hand edited) migrations should definitely go in the repo. Each migration has a pair of methods - one for moving to the new structure, one for going back to the old one. And yes, if you need to avoid breaking a non-null constraint, you'll need to include custom code in the migration.

**Comment by ageorgou** [comment]
Thanks @jonc. Added my current naive workaround for the migration, which however does not display existing rows nicely in the dashboard. Presumably the best way would be to:
- add the field without the constraint
- set the values of existing rows to the correct value
- add the non-null constraint
- create the index

But I'm not sure how worried we are about how existing hosts are displayed?

**Comment by jonc125** [comment]
Yeah, I don't like removing the uniqueness constraint as a workaround! That might lead SqlAlchemy to think it's there but the database not to have it, since the DB schema is created by applying the migrations sequentially.

Can you copy values from one column to another while creating? That might be a slightly quicker implementation in this case, especially since we don't have much data to migrate. But might be best just to do it properly :)

**Comment by ageorgou** [comment]
I don't think you can copy values while creating, hence the workaround, but I only had a relatively quick look.
To be clear, the uniqueness constraint was only in the index anyway! (or so I think)

**Comment by ageorgou** [comment]
Migration still needs to be cleared up, ran into difficulties/confusion with sqlalchemy.

**Comment by ageorgou** [comment]
Test seem to fail because of a connection error?
Edit: On second inspection, terraform fails to deploy the VM because of an error in the specification of the storage account. Perhaps an Azure change?
Edit^2: Error seems to be because of #52.

File: requirements/base.txt [hunk 1]
```diff
@@ -16,6 +16,7 @@ flask-sqlalchemy==2.2
 flask-sso==0.4.0
 flask-wtf==0.14.2
 flask==0.12.1
+haikunator==2.1.0
```
Comment (raquelalegre) [inline requirements/base.txt:19]: Local.txt must have a duplicated requirements.

File: src/cloudlabs/forms/add_host.py [hunk 2]
```diff
@@ -25,13 +25,14 @@ class AddHostForm(FlaskForm):
...
 
-    dns_name = StringField(
+    base_name = StringField(
         'URL',
         description='The canonical DNS name for this host, which must be '
                     'globally unique. Note that the .cloudlabs.rc.ucl.ac.uk '
                     'suffix will be added automatically, and should not be '
                     'included here.',
         validators=[v.Length(min=1, max=Host.dns_name.type.length)])
+    # TODO adapt max length to exclude suffixes/randomisation?
```
Comment (raquelalegre) [inline src/cloudlabs/forms/add_host.py:35]: Are TODOs like this one and below recorded on issues?

**Review by raquelalegre — APPROVED** [review]

File: requirements/base.txt (hunk 1 above)
Comment (raquelalegre) [reply requirements/base.txt:19]: Not needed!

File: src/cloudlabs/forms/add_host.py (hunk 2 above)
Comment (ageorgou) [reply src/cloudlabs/forms/add_host.py:35]: Noted on #1.

**Pull request merged**